File: java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;

public class Hamming {

    private int hammingDistance;

    public Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("strands must be of equal length");
        }

        this.hammingDistance = (int) IntStream.range(0, leftStrand.length())
            .filter(index -> leftStrand.charAt(index) != rightStrand.charAt(index)).count();
    }

    public int getHammingDistance() {
        return this.hammingDistance;
    }
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        if (leftStrand.length != rightStrand.length) {
            throw IllegalArgumentException("left and right strands must be of equal length")
        }

        var diff = 0

        for (i in 0..leftStrand.length - 1) {
            if (leftStrand[i] != rightStrand[i]) {
                diff++
            }
        }

        return diff

    }
}
